api/fbs/v1/schema.fbs

```diff
@@ -27,21 +27,50 @@ struct Duration {
     unit:DurationUint;
 }
 
+enum FieldType:byte { String = 0, Int = 1, StringArray = 2, IntArray = 3 }
+
+table FieldSpec {
+    name:string;
+    type:FieldType;
+}
+
+table TraceStateMap {
+    field:string;
+    val_success:string;
+    val_error:string;
+}
+
+// The key in TraceFieldMap are reserved by trace series engine. Their corresponding value is the Fields or 
+// the combination of Fields
+table TraceFieldMap {
+    // trace_id the unique identity of a single trace
+    trace_id:string;
+    // series_id groups entites into a storage bucket
+    series_id:[string];
+    // state indicates the trace is "success" or "error"
+    state:TraceStateMap;
+}
+
 // TraceSeries represents a trace storage object
 table TraceSeries {
     // metadata is the identity of a trace series
     metadata:Metadata;
+    // fields defines accetped fields
+    fields:[FieldSpec];
+    // reserved_fields_map indicates how to index reserved fields to ingested fields
+    reserved_fields_map:TraceFieldMap;
     // duration determines how long a TraceSeries keeps its data
     duration:Duration;
-    // updated_at indicates when the TraceSeries is updated
-    updated_at:uint64;
+    // updated_nanoseconds_at indicates when the TraceSeries is updated
+    updated_nanoseconds_at:uint64;
 }
 
 // Catalog refers to a placement contains objects belonged to a particular data type
 enum Catalog:byte { Trace = 0, Log = 1, Metric = 2 }
 
 // IndexType determine the index structure under the hood
-enum IndexType:byte { Text = 0, Numerical = 1, ID = 2}
+// Fields with SeriesInternal type is reserved by Series module, would not be indexed by Index module.
+enum IndexType:byte { Text = 0, Numerical = 1, ID = 2, MultiText = 3, MultiNumberical = 4, SeriesInternal = 5 }
 
 // IndexObject defines who should be indexed.
 table IndexObject {
```

Comment:
An indexName may be necessary for the object of `IndexRule`? Otherwise how can we determine the index while searching with API in `index.go`?

```go
type Repo interface {
	Search(indexName string, startTime, endTime uint64, conditions []Condition) ([]common.ChunkID, error)
}
```

Reply:
You can leverage methods in `IndexFilter` to get desired index rule.

Reply:
> You can leverage methods in `IndexFilter` to get desired index rule.

Oh, I see the metadata in the `IndexRule`. But do we need a `group` here? Or indexRule's `name` is globally unique?

Reply:
IndexObject is a portion of IndexRule instead of an individual resource. We don't set a group to it. 

IndexRule has metadata that has a group.

Reply:
> IndexObject is a portion of IndexRule instead of an individual resource. We don't set a group to it.
> 
> IndexRule has metadata that has a group.

Yes, I understood. But according to the signature of `Repo.Search`, we have to pass a `indexName` (string) as the first argument, specifically, which name should be given here?

Reply:
user will input the trace series, in the skywalking scenario, the name is `sw` and the group is `default`. The combination of them is the metadata of trace series. Now you have two paths to get the index name:

1. Using methods in `SchemaRepo`. Finding IndexRuleBinding by subject(above trace series) is the first step. Then you can get the desired IndexRule by the `rule_ref` in an IndexRuleBinding.

2. A sugar method in `IndexFilter`. `RulesBySubject` provides a similar querying function as indicated in the first option. You can use it to find the IndexRule by a TraceSeries's metadata.

The indexName of `Repo.Search` is the metadata of IndexRule. But I should twist the signature of this method to support a global index name.